packages/ember-glimmer/tests/integration/helpers/yield-test.js

```diff
@@ -0,0 +1,256 @@
+import { RenderingTest, moduleFor } from '../../utils/test-case';
+import { set } from 'ember-metal/property_set';
+import { Component } from '../../utils/helpers';
+
+moduleFor('Helpers test: {{yield}} helper', class extends RenderingTest {
+
+  ['@test a component with a layout set renders its template where the {{yield}} helper appears']() {
+    this.registerComponent('yield-comp', { template: '<div class="wrapper">{{yield}}</div>' });
+
+    this.render('{{#yield-comp}}<div class="page-body">page body</div>{{/yield-comp}}', { title: 'test' });
+
+    equal(this.$('div.wrapper div.page-body').length, 1, 'page-body is embedded within wrapping my-page');
+
+    this.assertStableRerender();
+  }
+
+  ['@test block should work properly even when templates are not hard-coded']() {
+    this.registerComponent('outer-comp', { template: '<div class="wrapper">{{yield}}</div>' });
+    this.registerComponent('inner-comp', { template: '{{#outer-comp}}<div class="page-body">Body</div>{{/outer-comp}}' });
+
+    this.render('{{inner-comp}}');
+
+    equal(this.$('div.wrapper div.page-body').length, 1, 'page-body is embedded within wrapping my-page');
+
+    this.assertStableRerender();
+  }
+
+  ['@test templates should yield to block, when the yield is embedded in a hierarchy of virtual views']() {
+    let list = [1, 2, 3, 4, 5];
+
+    this.registerComponent('outer-comp', { template: '<div class="times">{{#each list as |item|}}{{yield}}{{/each}}</div>' });
+
+    this.render('<div id="container">{{#outer-comp list=list}}<div class="times-item">Hello</div>{{/outer-comp}}</div>', { list: list });
+
+    let element1 = this.firstChild;
+
+    equal(this.$('div#container div.times-item').length, 5, 'times-item is embedded within wrapping container 5 times, as expected');
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'list', [6, 7]));
+
+    equal(this.$('div#container div.times-item').length, 2, 'times-item is embedded within wrapping container 5 times, as expected');
+
+    this.runTask(() => set(this.context, 'list', list));
+
+    let element2 = this.firstChild;
+
+    this.assertSameNode(element2, element1);
+  }
+
+  ['@test templates should yield to block, when the yield is embedded in a hierarchy of non-virtual views' ]() {
+    this.registerComponent('nested-comp', { template: '<div class="nesting">{{yield}}</div>' });
```

Comment:
This doesn't look like it's testing something different from other tests. :scissors: